packages/sdk/src/wallet/core/providers/smart/default/DefaultSmartWalletProvider.ts

```diff
@@ -50,27 +53,67 @@ export class DefaultSmartWalletProvider extends SmartWalletProvider {
 
   /**
    * Create a new smart wallet instance
-   * @description Creates a new smart wallet that will be deployed on first transaction.
-   * The wallet address is deterministically calculated from owners and nonce.
+   * @description Creates a new smart wallet and attempts to deploy it across all supported chains.
+   * The wallet address is deterministically calculated from owners and nonce. Deployment failures
+   * on individual chains do not prevent wallet creation - they are reported in the result.
    * @param owners - Array of wallet owners (addresses or WebAuthn public keys)
    * @param signer - Local account used for signing transactions
    * @param nonce - Optional nonce for address generation (defaults to 0)
-   * @returns Promise resolving to a new SmartWallet instance
+   * @param deploymentChainIds - Optional chain IDs to deploy the wallet to.
+   * If not provided, the wallet will be deployed to all supported chains. If you would
+   * like to lazily deploy the wallet on first transaction, you can provide an empty array.
+   * @returns Promise resolving to deployment result containing:
+   * - `wallet`: The created SmartWallet instance
+   * - `deploymentSuccesses`: Array of successful deployments with chainId and receipt
+   * - `deploymentFailures`: Array of failed deployments with chainId and error details
    */
   async createWallet(params: {
     owners: Signer[]
     signer: LocalAccount
     nonce?: bigint
-  }): Promise<DefaultSmartWallet> {
-    const { owners, signer, nonce } = params
-    return DefaultSmartWallet.create({
+    deploymentChainIds?: SupportedChainId[]
+  }): Promise<SmartWalletDeploymentResult<DefaultSmartWallet>> {
+    const { owners, signer, nonce, deploymentChainIds } = params
+    const wallet = await DefaultSmartWallet.create({
       owners,
       signer,
       chainManager: this.chainManager,
       lendProvider: this.lendProvider,
       nonce,
       attributionSuffix: this.attributionSuffix,
     })
+
+    const deploymentResults = await Promise.allSettled(
+      (deploymentChainIds ?? this.chainManager.getSupportedChains()).map(
```

Comment:
❓ We don't need to figure it out in this PR, but how do we want to handle the scenario:
1. VerbsConfig.chains == [1,2,3]
2. deploy some wallets
3. dev updates VerbsConfig.chains = [2,3,4] // removed + adds chain support. new wallet deployments needed

Comes with the challenge of building a stateless library that deals with a lot of state I guess.

Reply:
yeah, doing this in a stateless library is tricky. One thing that will somewhat help with this is that in the `getSmartWallet` function (which should be used by devs when they expect the wallet to already be deployed), the developer is expected pass in the `deploymentAddress` of the wallet.  If the  `deploymentAddress` they pass in does not correspond to a deployed smart wallet, then their txs will fail. This is not a great experience though, because the error isnt thrown when fetching the wallet, it would be thrown when they attempt to make a transaction with said wallet.

Reply:
will think through this edge case a bit more